tests/transforms/test_split_to_single_terms.py

```diff
@@ -0,0 +1,661 @@
+
+    @pytest.mark.parametrize(
+        "non_pauli_obs", [qml.Projector([0], wires=[1]), qml.Hadamard(wires=[1])]
+    )
+    def test_tape_with_non_pauli_obs(self, non_pauli_obs):
+        """Tests that the tape is split correctly when containing non-Pauli observables"""
+
+        measurements = [
+            qml.expval(qml.X(0) + qml.Y(0) + qml.X(1)),
+            qml.expval(non_pauli_obs + qml.Z(3)),
+        ]
+        tape = qml.tape.QuantumScript([qml.RX(1.2, 0)], measurements=measurements)
+
+        tapes, fn = split_to_single_terms(tape)
+
+        expected_tape = qml.tape.QuantumScript(
+            [qml.RX(1.2, 0)],
+            [
+                qml.expval(qml.X(0)),
+                qml.expval(qml.Y(0)),
+                qml.expval(qml.X(1)),
+                qml.expval(non_pauli_obs),
+                qml.expval(qml.Z(3)),
+            ],
+        )
+
+        assert qml.equal(tapes[0], expected_tape)
+
+        result = [[0.1, 0.2, 0.3, 0.4, 0.5]]
+        assert fn(result) == ((0.1 + 0.2 + 0.3), (0.4 + 0.5))
+
+    @pytest.mark.parametrize(
+        "observable",
+        [
+            qml.X(0) + qml.Y(1),
+            2 * (qml.X(0) + qml.Y(1)),
+            3 * (2 * (qml.X(0) + qml.Y(1)) + qml.X(1)),
+        ],
+    )
+    def test_splitting_sums_in_unsupported_mps_raises_error(self, observable):
+
+        tape = qml.tape.QuantumScript([qml.X(0)], measurements=[qml.counts(observable)])
+        with pytest.raises(
+            RuntimeError, match="Cannot split up terms in sums for MeasurementProcess"
+        ):
+            _, _ = split_to_single_terms(tape)
+
+
+@patch(
+    "pennylane.devices.default_qubit.observables",
+    original_observables - {"Sum", "Hamiltonian", "LinearCombination"},
+)
+class TestIntegration:
+    """Tests the ``split_to_single_terms`` transform performed on a QNode. In these tests,
+    the supported observables of ``default_qubit`` are mocked to make the device reject Sum,
+    Hamiltonian and LinearCombination, to ensure the transform works as intended."""
+
+    def test_splitting_sums(self):
+        """Test that the transform takes a tape that is not executable on a device that
+        doesn't support Sum/Hamiltonian, and turns it into one that is"""
+
+        coeffs, obs = [0.1, 0.2, 0.3, 0.4, 0.5], single_term_obs_list
+
+        if not qml.operation.active_new_opmath():
+            obs = _convert_obs_to_legacy_opmath(obs)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(1.2, wires=0)
+            return qml.expval(qml.Hamiltonian(coeffs, obs))
+
+        @split_to_single_terms
+        @qml.qnode(dev)
+        def circuit_split():
+            qml.RX(1.2, wires=0)
+            return qml.expval(qml.Hamiltonian(coeffs, obs))
+
+        with pytest.raises(qml.DeviceError, match="not supported on default.qubit"):
+            circuit()
+
+        circuit_split()
```

Comment:
```suggestion
        with dev.tracker:
            circuit_split()
        assert dev.tracker.totals['simulations'] == 1
```
This helps us double check that it doesn't get split up into multiple circuits.